lib/enumerable/any.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_any?
+    raise 'Please provide a block' unless block_given?
+
```

Comment:
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

Reply:
```ruby
def my_any?(arg = nil)
    if block_given?
      my_each { |i| return true if yield(i) }
    elsif arg.nil?
      my_each { |i| return true if i }
    else
      my_each { |i| return true if i.is_a?(arg) }
    end
    false
  end
```

Solved